### ISSUE TITLE
Create backup task

### ISSUE THREAD

**Issue by sverhoeven** [research-software-directory/research-software-directory#201]
Create a new task in  tasks/ that will make a dump (`mongodump -d rsd -o /dump --excludeCollection=commit --excludeCollection=software_cache --excludeCollection=release`) an copy it using xenon-cli (`xenon filesystem webdav ... upload mongdump-<timestamp>.tar.gz ....`).

Task should run once a day and that the xenon command is configurable using a env var.

**Comment by sverhoeven** [comment]
After task has been made the `config/backup.sh` can be removed.

**Comment by jspaaks** [comment]
Actually, I'd like to get your opinion on a slightly different setup:

First, the setup to copy the db files out of the container:
- add a file with the following content to ``database/``, maybe call it ``backup.sh``
    ```
    mongodump --db rsd --out /dump \
    --excludeCollection=commit \
    --excludeCollection=software_cache \
    --excludeCollection=release
    ```
- add a COPY to the Dockerfile that copies ``backup.sh`` into the image
- add a docker volume ``db-dump`` or something to ``docker-volumes``; let it map to ``/dump``
- start a cronjob in the database container such that the backup runs every 24 h. 

Second, 
- create a new directory/service named ``backup``, which has a Dockerfile that installs the necessary stuff for using ``xenon-cli``. 
- mount the docker volume ``docker-volumes/db-dump`` to this container.
- Add secrets to the ``.env`` / ``secrets`` file to gain access to a backup location. 
- add a cron job that starts 5 minutes later than the cron job in the ``database`` container

Related but not strictly necessary:
- rename ``tasks`` service/container/directory to ``harvesting``, which is more precise/ more descriptive 
- in tasks/app, rename ``--task`` to ``--harvest`` with an identifier of what is being harvested: e.g.
    - ``--harvest commits``
    - ``--harvest mentions``
    - ``--harvest releases`` (``citations``?)
    - ``--harvest oaipmh`` (``metadata``?)
- in tasks/app, make a new command with no arguments, something like ``--resolve-foreign-keys``to replace what is now ``--task cache_software``

@sverhoeven and @Tommos0 let me know if you like this plan, if you do i'll get on it right away.

**Comment by sverhoeven** [comment]
Nicely worked out. 
Timing the 2 cronjobs to happen after each other seems more complicated than needed. Lets do mongodump in the backup service.

**Comment by jspaaks** [comment]
OK, implemented most of the stuff above in the 2 PRs #209 and #210.
